### PR TITLE
Try around fixing #1836

### PR DESCRIPTION
**Pull request by matthid**

### CODE REVIEW

File: src/fable-library/Util.ts [hunk 1]
```diff
@@ -559,29 +561,50 @@ export function uncurry(arity: number, f: Function) {
...
+
+  return t;
 }
 
 export function curry(arity: number, f: Function): Function {
   if (f == null) { return null; }
+
+  let t :any = f;
+  if (!isUndefined(t.fable_curried)) {
+    if (t.fable_arity >= arity) {
```
Comment (matthid) [inline src/fable-library/Util.ts]: This check is probably wrong. Even when `arity > fable_arity` we probably want to use the `fable_curried` field. However, the code for this is probably a bit tricky (well at least we only support arities up to 8)?

File: src/fable-library/Util.ts [hunk 2]
```diff
@@ -559,29 +561,50 @@ export function uncurry(arity: number, f: Function) {
...
+  t.fable_arity = arity;
+
+  return t;
 }
 
 export function curry(arity: number, f: Function): Function {
   if (f == null) { return null; }
+
+  let t :any = f;
+  if (!isUndefined(t.fable_curried)) {
```
Comment (vbfox) [inline src/fable-library/Util.ts]: Small question, what is `isUndefined` and how does it differ from `=== undefined` ?

Comment (matthid) [reply src/fable-library/Util.ts]: To be honest I don't know. It just sounds like the right thing to do ;)

**Comment by matthid** [comment]
Fixed in master in a similar way :)

**Pull request closed without merging**